script.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * SmartModalDialog creates a dialog with a group of buttons as select
+ * options. It also includes open/close animation and bubbling animation.
+ * The template is shown below:
+ *   <smart-dialog class="modal-dialog">
+ *     <div class="outer-container">
+ *       <div class="container">
+ *         <div class="modal-dialog-message">
+ *         </div>
+ *         <div class="modal-dialog-button-group">
+ *           <smart-button></smart-button>
+ *           ...
+ *           <smart-button></smart-button>
+ *         </div>
+ *       </div>
+ *     </div>
+ *   </smart-dialog>
+ */
+
+(function(exports) {
+  'use strict';
+
+  function SmartModalDialog() {
+
+    // Determine whether this dialog is opened or not
+    this.isOpened = false;
+    this._clickedIndex = null;
+
+    this.smartBubble = document.createElement('smart-bubbles');
+    document.body.appendChild(this.smartBubble);
```

Comment:
I thought you have to wrap what you want to bubble _inside_ `smart-bubble` rather than put it randomly in a document